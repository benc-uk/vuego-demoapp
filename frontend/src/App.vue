<!--
Core app component
----------------------------------------------
// Ben C, April 2018, Updated for Vue3 2021
-->

<template>
  <div>
    <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
      <div class="container-fluid">
        <a class="navbar-brand text-light" href="#">Vue &amp; Go Demo</a>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div id="navbarSupportedContent" class="collapse navbar-collapse">
          <ul class="navbar-nav me-auto mb-2 mb-lg-0">
            <li class="nav-item">
              <router-link class="nav-link active" to="/home"> <i class="fas fa-home fa-fw"></i> Home</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/info"> <i class="fas fa-cog fa-fw"></i> Info</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/monitor"> <i class="fas fa-tachometer-alt fa-fw"></i> Monitor</router-link>
            </li>
            <li v-if="weatherEnabled" class="nav-item">
              <router-link class="nav-link active" to="/weather"> <i class="fas fa-umbrella fa-fw"></i> Weather</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/about"> <i class="fas fa-info-circle fa-fw"></i> About</router-link>
            </li>
          </ul>
          <div v-if="authEnabled" class="d-flex">
            <button v-if="!user" class="btn btn-info btn-lg" variant="light" @click="doLogin()">
              <i class="fas fa-sign-in-alt"></i>&nbsp; Login
            </button>
            <router-link v-if="user" class="btn btn-info" to="/user" variant="light"> <i class="fas fa-user"></i>&nbsp; User </router-link>
          </div>
        </div>
      </div>
    </nav>

    <div class="container mt-5">
      <router-view v-slot="{ Component }">
        <transition name="route" mode="out-in">
          <component :is="Component" />
        </transition>
      </router-view>
    </div>
  </div>
</template>

<script>
import auth from './services/auth'
import router from './router'
import { config } from './main'

export default {
  data: function () {
    return {
      user: null,
      // We switch the user sign-in feature on or off depending if auth has been configured
      authEnabled: auth.isConfigured(),
      weatherEnabled: config.weatherEnabled
    }
  },

  async created() {
    // Restore any cached or saved local user
    if (auth.isConfigured()) {
      this.user = auth.user()
    }
  },

  methods: {
    async doLogin() {
      try {
        await auth.login()
        this.user = auth.user()
        router.push('user')
      } catch (err) {
        this.error = err.toString()
      }
    },

    clearUser() {
      this.user = null
    }
  }
}
</script>

<style>
html {
  min-height: 100%;
}

.route-enter-from {
  opacity: 0;
  transform: translateX(100px);
}
.route-enter-active {
  transition: all 0.3s ease-out;
}
.route-leave-to {
  opacity: 0;
  transform: translateX(-100px);
}
.route-leave-active {
  transition: all 0.3s ease-in;
}
</style>
