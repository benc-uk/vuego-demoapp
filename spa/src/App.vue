<!--
Core app component, just template no code/logic
----------------------------------------------
Ben C, April 2018
-->

<template>
  <div id="app">
    <b-navbar toggleable="md" type="dark" variant="primary">
      <b-navbar-toggle target="nav_collapse" />
      <b-navbar-brand to="/"> <img src="./assets/vuejs.svg" height="50px" />Vue.js &amp; Go </b-navbar-brand>
      <b-collapse id="nav_collapse" is-nav>
        <b-navbar-nav>
          <b-button to="/home" variant="dark"> <fa icon="home" />&nbsp; Home </b-button>
          <b-button to="/info" variant="dark"> <fa icon="cogs" />&nbsp; Info </b-button>
          <b-button to="/monitor" variant="dark"> <fa icon="tachometer-alt" />&nbsp; Monitor </b-button>
          <b-button to="/weather" variant="dark"> <fa icon="umbrella" />&nbsp; Weather </b-button>
          <b-button to="/about" variant="dark"> <fa icon="info-circle" />&nbsp; About </b-button>
        </b-navbar-nav>

        <b-navbar-nav v-if="authEnabled" class="ml-auto">
          <b-button v-if="!user" variant="light" @click="doLogin()"> <fa icon="sign-in-alt" />&nbsp; Login </b-button>
          <b-button v-if="user" to="/user" variant="light"> <fa icon="user" />&nbsp; User </b-button>
        </b-navbar-nav>
      </b-collapse>
    </b-navbar>

    <br />

    <b-container>
      <transition name="slidedown">
        <router-view @logoutEvent="clearUser()" />
      </transition>
    </b-container>
  </div>
</template>

<script>
import auth from './services/auth'
import router from './router'

export default {
  data: function() {
    return {
      user: {},
      authEnabled: auth.isConfigured()
    }
  },

  async created() {
    // Restore any cached or saved local user
    if (auth.isConfigured()) {
      this.user = auth.user()
    }
  },

  methods: {
    async doLogin() {
      try {
        await auth.login()
        this.user = auth.user()
        router.push('user')
      } catch (err) {
        this.error = err.toString()
      }
    },

    clearUser() {
      this.user = null
    }
  }
}
</script>

<style>
html {
  font-size: 20px;
}
#app,
html,
body {
  height: 100%;
  background-color: #317256;
}
.b-navbar-nav {
  color: red;
}
.slidedown-enter-active {
  transition: transform 0.4s ease-out;
  transform-origin: top;
}
.slidedown-enter {
  transform: scaleY(0);
}
</style>

<style scoped>
.btn {
  margin-right: 0.3rem;
  width: 7rem;
}
</style>
