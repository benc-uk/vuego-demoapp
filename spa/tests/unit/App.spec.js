import { createLocalVue, shallowMount } from '@vue/test-utils'
import App from '@/App.vue'
import flushPromises from 'flush-promises'
import VueRouter from 'vue-router'
//import { toHaveRouteName } from 'vue-test-utils-helpers'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('App.vue', () => {
  it('renders main app screen', async () => {
    const wrapper = shallowMount(App, { localVue, router })

    wrapper.find('b-button[to="/home"]').trigger('click')
    await wrapper.vm.$nextTick()
    await flushPromises()

    //console.log(wrapper.vm.$route)
    //expect(wrapper.vm.$route.name).toBe('home')

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})
