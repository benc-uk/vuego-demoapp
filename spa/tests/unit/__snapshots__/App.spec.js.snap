// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders main app screen 1`] = `
<div id="app">
  <b-navbar toggleable="md" type="dark" variant="primary">
    <b-navbar-toggle target="nav_collapse"></b-navbar-toggle>
    <b-navbar-brand to="/"><img src="./assets/vuejs.svg" height="50px">Vue.js &amp; Go
    </b-navbar-brand>
    <b-collapse id="nav_collapse" is-nav="">
      <b-navbar-nav>
        <b-button to="/home" variant="dark">
          <fa icon="home"></fa>&nbsp; Home
        </b-button>
        <b-button to="/info" variant="dark">
          <fa icon="cogs"></fa>&nbsp; Info
        </b-button>
        <b-button to="/monitor" variant="dark">
          <fa icon="tachometer-alt"></fa>&nbsp; Monitor
        </b-button>
        <b-button to="/weather" variant="dark">
          <fa icon="umbrella"></fa>&nbsp; Weather
        </b-button>
        <b-button to="/about" variant="dark">
          <fa icon="info-circle"></fa>&nbsp; About
        </b-button>
      </b-navbar-nav>
    </b-collapse>
  </b-navbar> <br>
  <b-container>
    <transition-stub name="slidedown">
      <router-view-stub name="default"></router-view-stub>
    </transition-stub>
  </b-container>
</div>
`;
